applications/server/controllers/proc_harness.py

```diff
@@ -12,62 +12,46 @@
-# Do we have to worry about stuff coming in at the same time? Currently a bug
-def get_procedure_data(procedure_id, stable):
-    max = revisions_table.last_update.max()
-    if stable:
-        date = db((revisions_table.procedure_id == procedure_id) &
-                                      (revisions_table.stable_version == stable)).select(max).first()[max]
-    else:
-        date = db(revisions_table.procedure_id == procedure_id).select(max).first()[max]
-    return db((revisions_table.procedure_id == procedure_id) &
-              (revisions_table.last_update == date)).select(revisions_table.procedure_data).first().procedure_data
+def get_procedure_status(device_id):
+    """
+    Called upon client request for procedure info
 
-# True for stable is save stable, False is save temp
-# when save is stable, flush all temp versions
-def save(procedure_id, procedure_data, stable):
-    # insert new record
-    revisions_table.insert(procedure_id = procedure_id,
-                           procedure_data = procedure_data,
-                           last_update = datetime.datetime.utcnow(),
-                           stable_version = stable)
-    #flush temp versions for stable save
-    if stable:
-        db((db[table].uuid == item.uuid) &
-           (db[table].id != item.id)).delete()
-        db((revisions_table.procedure_id == procedure_id) &
-           (revisions_table.stable_version == False)).delete()
+    :param device_id: Device id for which the procedure sync is being performed
+    :type device_id: long
+    :return Dict of the format {procedure_id: last_update_stable}
+                    that specifies the last save time for each procedure associated with the device
+    :rtype:
+    """
 
-####### API FOR PROCEDURE HARNESS TEAM ##########
+    # Get all procedure_ids for the device_id
+    procedure_ids = db((devices.device_id == device_id) &
+                       (devices.user_email == proc_table.user_email)).select(proc_table.procedure_id)
 
-# called in response to client request for procedure info
-# returns a dictionary of the format {procedure_id, last_update_stable}
-def get_procedure_status(device_id):
-    # 1. Get all procedure_ids for the device_id
-    procedure_ids = db((db.devices.device_id == device_id) &
-                       (db.devices.user_id == proc_table.user_id)).select(proc_table.procedure_id)
-    # 2. build dictionary containing last_update_stable date for each procedure_id
+    # Build dictionary containing last_update date for each procedure_id
     procedure_info = {}
     for proc in procedure_ids:
-        procedure_info[proc.procedure_id] = get_procedure_data(proc, True)
-    # 3. return dictionary
-    return procedure_info
\ No newline at end of file
+        pid = proc.procedure_id
+        procedure_info[pid] = db(revisions_table.procedure_id == pid).select(revisions_table.last_update).first().last_update
+
+    return procedure_info
+
+
+def get_procedure_update(procedure_id_lst):
```

Comment:
I thought that this would be called with a device_id, and would return the list of all procedures for that device_id with their date.